packages/entity-server/src/models/Entity.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { EntityModel as BaseEntityModel, EntityType as BaseEntityType } from '@tupaia/database';
+import { Model } from './types';
+
+export interface EntityFields {
+  readonly code: string;
+  readonly country_code: string;
+}
+
+export interface EntityType extends BaseEntityType, EntityFields {
+  getChildren: (hierarchyId: string) => Promise<EntityType[]>;
+  getChildrenViaHierarchy: (hierarchyId: string) => Promise<EntityType[]>;
```

Comment:
I've just looked at this second method, and I'm not sure it has any place in modern Tupaia? Is there any distinction between the two? I can see that the second would exclude any canonically connected descendants, but I'm not sure that's a use case that's important?

Reply:
I'll have to re-test this, at the time I added it since it seems the common way that we fetch children of project entities throughout Tupaia. I was failing to get the expected results using `getChildren()` but I think that was actually due to me incorrectly using `hierarchyName` instead of `hierarchyId`. Good pickup, I'd prefer not to use this one if possible...